core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy

```diff
@@ -0,0 +1,573 @@
+            newHighBoundary = other.highBoundary
+        } else if (high && !other.high) {
+            newHigh = high
+            newHighBoundary = highBoundary
+        } else if (!high && !other.high) {
+            newHigh = null
+            newHighBoundary = highBoundary
+        } else {
+            switch (compare(high, other.high)) {
+                case 1:
+                    newHigh = other.high
+                    newHighBoundary = other.highBoundary
+                    break
+                case 0:
+                    newHighBoundary = (highBoundary == EXCLUSIVE || other.highBoundary == EXCLUSIVE) ? EXCLUSIVE : INCLUSIVE
+                    newHigh = high
+                    break
+                case -1:
+                    newHigh = high
+                    newHighBoundary = highBoundary
+            }
+        }
+        return new GemVersion(newLowBoundary, newLow, newHigh, newHighBoundary)
+    }
+
+    /** Creates a new GEM version requirement which that the lowest of two requirements and the highest
+     * of those same requirement
+     *
+     * @param other Other GEM to combine with
+     * @return New GEM version requirement.
+     *
+     * @since 2.0
+     */
+    GemVersion union(GemVersion other) {
+        List<GemVersion> pair = [this, other]
+        GemVersion min = pair.min()
+        GemVersion max = pair.max()
+
+        new GemVersion(min.lowBoundary, min.low, max.high, max.highBoundary)
+    }
+
+    /** Allows for versions to be compared and sorted.
+     *
+     * @param other Other GEM version to compare to.
+     * @return -1, 0 or 1.
+     *
+     * @since 2.0
+     */
+    @Override
+    int compareTo(GemVersion other) {
+        int loCompare = compare(low, other.low)
+        if (loCompare) {
+            return loCompare
+        }
+
+        if (lowBoundary != other.lowBoundary) {
+            if (lowBoundary == OPEN_ENDED) {
+                return -1
+            } else if (other.lowBoundary == OPEN_ENDED) {
+                return 1
+            }
+            return lowBoundary == INCLUSIVE ? -1 : 1
+        }
+
+        int hiCompare = compare(high, other.high)
+
+        if (hiCompare) {
+            return hiCompare
+        }
+
+        if (highBoundary != other.highBoundary) {
+            if (highBoundary == OPEN_ENDED) {
+                return 1
+            } else if (other.highBoundary == OPEN_ENDED) {
+                return -1
+            }
+            return highBoundary == INCLUSIVE ? 1 : -1
+        }
+
+        0
+    }
+
+    /**
+     * examines the version range on conflict, i.e. lower bound bigger then
+     * upper bound.
+
+     * @return boolean true if lower bound bigger then upper bound
+     */
+    boolean conflict() {
+        compare(stripNonIntegerTail(low), stripNonIntegerTail(high)) == 1
+    }
+
+    /** String of the underlying data as Ivy version range.
+     *
+     * @return Gradle Ivy version range
+     */
+    String toString() {
```

Comment:
This has changes to rather print an equivalent Ivy version range rather than a Maven one. This reflects closer to how Gradle specifies non-fixed versions.